### DISCUSSION TITLE
What is the best way to increase and decrease the volume with an encoder?

### DISCUSSION THREAD

**Discussion by timeylies** [pschatzmann/arduino-audiokit#43] [Q&A] [answered]
So I'm trying to make a Bluetooth receiver with an LCD and a rotary encoder and each time I try to add or subtract 1 to the volume of the kit and it never changes

**Comment by pschatzmann** [comment]
Which API are you using ? In the AudioKitStream the volume is between 0 and 1.0!

**Reply by timeylies** [reply]
I'm using the AudioKitStream

**Reply by pschatzmann** [reply]
So, adding 1 does not make any sense: this is already the max value!

**Reply by timeylies** [reply]
I mean going 1 to 100

**Reply by pschatzmann** [reply]
But the range is from 0.0 to 1.0 ! If your range is from 1 to 100 you need to divide by 100! Don't forget to convert the value to float first before you divide...

**Reply by timeylies** [reply]
Yes I am doing that but problem is that there's not a good way to turn up or down the volume: I have to do this to change volume

`kit.volume(kit.volume() + 0.01) `

**Reply by pschatzmann** [reply]
If you print out the value you will see what you do wrong! I can only repeat myself: Don't forget to convert the value to float first before you divide! 1 / 100 = 0; 1.0 / 100 = 0.01

**Reply by timeylies** [reply]
Yea but that isn't the problem, what I meant is that there should be two functions to increase or decrease the volume of the kit

**Comment by pschatzmann** [comment]
No, I disagree! 
The one existing incremental function is sufficient: Increasing is with positive numbers, decreasing is with negative numbers.
There are actually the 2 button handler methods for volume up and volume down that you could call alternatively. But they are less flexible...

Did you look into the class documentation if the [AudioKitStream](https://pschatzmann.github.io/arduino-audio-tools/html/classaudio__tools_1_1_audio_kit_stream.html) class ? I really created the documentation to avoid this kind of discussions...

**Comment by timeylies** [comment]
Yes I did but how can I do it

Can you show me an example

yariel

**Comment by pschatzmann** [comment, marked as answer]
Just call kit.incrementVolume(2) to increase or kit.incrementVolume(-2) to decrease the volue - or use whatever number fits you. The bigger the number the bigger the impact... 
You should really learn to read the documentation!

**Reply by timeylies** [reply]
Yea true imma get on reading the docs but thank you
